### DISCUSSION TITLE
how to use spritemap texture atlases

### DISCUSSION THREAD

**Discussion by wormylmao** [HaxeFlixel/flixel#2395] [Q&A]
just to make maxrects to a whole new level
![image](https://user-images.githubusercontent.com/82244539/131192764-4b59d0e4-c398-4f3f-97ff-9ded1e538572.png)

**Comment by TheHENOOB** [comment]
are those aseprite .json data? if not, so what is those .json files?

**Reply by wormylmao** [reply]
adobe texture atlas

**Reply by wormylmao** [reply]
https://youtu.be/yr9iHJlbezo?t=69

**Reply by TheHENOOB** [reply]
why you don't use [Sparrow](https://ninjamuffin99.newgrounds.com/news/post/1090580) instead of adobe texture atlas? i don't think haxeflixel support this format yet

**Reply by wormylmao** [reply]
its used in alot of haxeflixel games

**Reply by TheHENOOB** [reply]
are you mean about adobe texture atlas?

**Reply by wormylmao** [reply]
yes

**Reply by TheHENOOB** [reply]
do you know those games? maybe some of them are open source so you can check out how to use the atlas

**Reply by TheHENOOB** [reply]
also can i check the contents of the .json file so i can confirm it supports haxeflixel?

**Reply by wormylmao** [reply]
yeah sure

**Reply by wormylmao** [reply]
**SPRITEMAP.JSON**
{"ATLAS": {"SPRITES":[ 
{"SPRITE" : {"name": "0000","x":0,"y":363,"w":119,"h":118,"rotated": true}},
{"SPRITE" : {"name": "0001","x":0,"y":0,"w":363,"h":363,"rotated": false}}
]},
"meta": {
"app": "Adobe Animate",
"version": "21.0.6.41649",
"image": "spritemap1.png",
"format": "RGBA8888",
"size": {"w":363,"h":481},
"resolution": "1"
}
}

**ANIMATION.JSON**

"AN": {
"N": "WEIRDGAME",
"SN": "madcircleboss",
"TL": {
"L": [
{
"LN": "Layer_2",
"FR": [
{
"I": 0,
"DU": 1,
"E": [
{
"SI": {
"SN": "madveins",
"IN": "",
"ST": "G",
"FF": 0,
"LP": "LP",
"TRP": {
"x": 53.75,
"y": 53.75
},
"M3D": [
0.9659271240234375,
0.2587738037109375,
0.0,
0.0,
-0.2587738037109375,
0.9659271240234375,
0.0,
0.0,
0.0,
0.0,
1.0,
0.0,
236.65,
41.45,
0.0,
1.0
]
}
}
]
}
]
},
{
"LN": "Layer_1",
"FR": [
{
"I": 0,
"DU": 1,
"E": [
{
"SI": {
"SN": "bosscircle",
"IN": "",
"ST": "G",
"FF": 0,
"LP": "LP",
"TRP": {
"x": 165.85,
"y": 165.85
},
"M3D": [
1.0,
0.0,
0.0,
0.0,
0.0,
1.0,
0.0,
0.0,
0.0,
0.0,
1.0,
0.0,
0.0,
0.0,
0.0,
1.0
]
}
}
]
}
]
}
]
}
},
"SD": {
"S": [
{
"SN": "madveins",
"TL": {
"L": [
{
"LN": "Layer_2",
"FR": [
{
"I": 0,
"DU": 1,
"E": [
{
"ASI": {
"N": "0000",
"M3D": [
1.0,
0.0,
0.0,
0.0,
0.0,
1.0,
0.0,
0.0,
0.0,
0.0,
1.0,
0.0,
-5.0,
-5.0,
0.0,
1.0
]
}
}
]
}
]
}
]
}
},
{
"SN": "bosscircle",
"TL": {
"L": [
{
"LN": "Layer_2",
"FR": [
{
"I": 0,
"DU": 1,
"E": [
{
"ASI": {
"N": "0001",
"M3D": [
1.0,
0.0,
0.0,
0.0,
0.0,
1.0,
0.0,
0.0,
0.0,
0.0,
1.0,
0.0,
-16.0,
-16.0,
0.0,
1.0
]
}
}
]
}
]
}
]
}
}
]
},
"MD": {
"FRT": 24.0
}
}

**Comment by ninjamuffin99** [comment]
if these are adobe animate spritesheet exports, right now HaxeFlixel doesn't have proper support for them, you would have to implement that yourself!

**Reply by wormylmao** [reply]
so i cant use those texture atlases?

**Reply by wormylmao** [reply]
i guess i gotta stick to sparrow
